src/presign/participant.rs

```diff
@@ -926,10 +926,10 @@ impl PresignKeyShareAndInfo {
         };
 
         let r1_private = round_one::Private {
-            k,
-            rho,
+            k: k.into(),
+            rho: rho.into(),
```

Comment:
I am not sure if this is the right place to convert from a `Nonce` to a `SecretNonce`. For example, `rho` is generated above in the `EncryptionKey::encrypt` function, this function create a regular `Nonce`. The `encrypt` function uses the nonce like so `let c = self.encrypt_with_nonce(x, &MaskedNonce(nonce.clone()))?;`. Here, we are cloning the nonce. Since this is not a `SecretNonce` it will go out of scope without getting zeroized. So we have a copy of `rho` lingering in memory.

I believe we need to handle these cases as well? Otherwise, what is the point of zeroizing here but not there?